test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,355 @@ package provisioner
 import (
 	"context"
 
+	"errors"
+	"fmt"
+	"os"
 	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/containerservice/mgmt/containerservice"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+	log "github.com/sirupsen/logrus"
+	kconf "sigs.k8s.io/e2e-framework/klient/conf"
 )
 
 func init() {
 	newProvisionerFunctions["azure"] = NewAzureCloudProvisioner
 }
 
-// AzureCloudProvisioner implements the CloudProvisioner interface for Azure.
+func createResourceGroup() error {
+	newRG := resources.Group{
+		Location: &AzureProps.Location,
+	}
+
+	log.Infof("Creating Resource group %s.\n", AzureProps.ResourceGroupName)
+	resourceGroup, err := AzureProps.ResourceGroupClient.CreateOrUpdate(context.Background(), AzureProps.ResourceGroupName, newRG)
+	if err != nil {
+		fmt.Println("Failed to create resource group:", err)
+		return err
```

Comment:
Add context to the error? I would do something like `fmt.Errorf("creating resource group: %w", err)`